core/src/main/java/com/linecorp/armeria/client/endpoint/PropertiesEndpointGroup.java

```diff
@@ -181,14 +247,11 @@ private static void validateDefaultPort(int defaultPort) {
                       "defaultPort: %s (expected: 1-65535)", defaultPort);
     }
 
-    private final List<Endpoint> endpoints;
-
-    private PropertiesEndpointGroup(List<Endpoint> endpoints) {
-        this.endpoints = endpoints;
-    }
-
     @Override
-    public List<Endpoint> endpoints() {
-        return endpoints;
+    public void close() {
+        super.close();
```

Comment:
Maybe we should wrap with `try-finally` and call `closeCallback` in the `finally` block?